packages/netsuite-adapter/src/filters/workflow_account_specific_values.ts

```diff
@@ -0,0 +1,772 @@
+    .map(res => res[0])
+    // 0 (zero) can be used in a formula (e.g 'arrayIndexOf(...) < 0'), but it's not an internalid
+    .filter(internalId => internalId !== '0')
+  if (params.length !== internalIds.length) {
+    log.warn(
+      'params length %d do not match the internal ids extracted from the formula: %s',
+      params.length,
+      formulaWithInternalIds
+    )
+    return
+  }
+  _.sortBy(params, param => value[INIT_CONDITION][FORMULA].indexOf(`"${param.name}"`))
+    .forEach((param, index) => {
+      if (param.value === ACCOUNT_SPECIFIC_VALUE) {
+        const internalId = internalIds[index]
+        const paramType = getQueryRecordFieldType(
+          instance,
+          param,
+          'value',
+          suiteQLTablesMap,
+          selectRecordTypeMap,
+        )
+        if (paramType !== undefined) {
+          setFieldValue({ name: 'value', type: paramType }, param, internalId, suiteQLTablesMap)
+        }
+      }
+    })
+}
+
+const setValuesInInstance = (
+  instance: InstanceElement,
+  record: QueryRecord,
+  results: QueryRecordResponse[],
+  suiteQLTablesMap: Record<string, InstanceElement>,
+  selectRecordTypeMap: Record<string, unknown>
+): void => {
+  const innerValue = record.elemID.isTopLevel()
+    ? instance.value
+    : resolvePath(instance, record.elemID)
+  const innerResult = getInnerResult(results, record.path)
+  if (innerResult === undefined) {
+    log.warn('missing record %s in path %s: %o', record.serviceId, record.path.join('.'), results)
+    return
+  }
+  record.fields.forEach(field => {
+    const internalId = innerResult.body[field.name]
+    if (internalId === undefined || internalId === '') {
+      log.warn('missing field %s in record %s: %o', field.name, record.serviceId, innerResult.body)
+      return
+    }
+    if (typeof internalId !== 'string') {
+      log.warn('field %s in record %s is not a string: %o', field.name, record.serviceId, internalId)
+      return
+    }
+    if (field.type === FORMULA_PARAMS) {
+      setParametersValues(instance, innerValue, internalId, suiteQLTablesMap, selectRecordTypeMap)
+    } else {
+      setFieldValue(field, innerValue, internalId, suiteQLTablesMap)
+    }
+  })
+}
+
+const getSuiteQLNameToInternalIdsMap = async (
+  elementsSource: ReadOnlyElementsSource
+): Promise<Record<string, Record<string, string[]>>> => {
+  const suiteQLTableInstances = await awu(await elementsSource.list())
+    .filter(elemId => elemId.idType === 'instance' && elemId.typeName === SUITEQL_TABLE)
+    .map(elemId => elementsSource.get(elemId))
+    .filter(isInstanceElement)
+    .toArray()
+
+  return _(suiteQLTableInstances)
+    .keyBy(instance => instance.elemID.name)
+    .mapValues(instance => instance.value[INTERNAL_IDS_MAP] as InternalIdsMap)
+    .mapValues(
+      internalIdsMap => _(internalIdsMap)
+        .entries()
+        .groupBy(row => row[1].name)
+        .mapValues(rows => rows.map(row => row[0]))
```

Comment:
I think this is better in terms of readibility
```suggestion
        .groupBy((_key, value) => value.name)
        .mapValues(rows => rows.map((key, _value) => key))
```